### ISSUE TITLE
Create categories field block

### ISSUE THREAD

**Issue by louwie17** [woothemes/woocommerce#37263]
**Description**

Current blocked by: https://github.com/woocommerce/woocommerce/issues/37043

Users should have the ability to add categories to a product within the new product block editor.
This issue would involve creating a new block for categories, we should be able to re-use a large part of the current category field displayed in the product editor.

Be aware that there is currently some work being done on making use of the new TreeControl component within the category field: https://github.com/woocommerce/woocommerce/issues/37043 this may or may not be merged prior to starting this work.

The category block is displayed in its own section, this will be part of this work, as far as just adding this to the template.json. Adding the section icon will be done as part of a separate issue.

**Design**

![Image](https://user-images.githubusercontent.com/2240960/225603324-dc53f7f4-a105-4b70-8a6d-49328b35ac14.png)

AkVNGImLgSqCObTQ3idVn7-fi-10-139

**Acceptance criteria**

- [ ] A category block should be available in the product block editor
- [ ] Users should be able to search and select categories and have it save correctly
- [ ] Category block should be displayed within a new section **Organization & visibility**

**Comment by louwie17** [comment]
This has already been done!
